jsonrpc/namespaces/eth/filters/api.go

```diff
@@ -0,0 +1,679 @@
+}
+
+// Backend defines the methods requided by the PublicFilterAPI backend
+type Backend interface {
+	GetBlockByNumber(blockNum rpc.BlockNumber, fullTx bool) (map[string]interface{}, error)
+	HeaderByNumber(blockNum rpc.BlockNumber) (*ethtypes.Header, error)
+	HeaderByHash(blockHash common.Hash) (*ethtypes.Header, error)
+	GetLogs(blockHash common.Hash) ([][]*ethtypes.Log, error)
+	GetLogsByHeight(*int64) ([][]*ethtypes.Log, error)
+	BlockBloom(blockRes *coretypes.ResultBlockResults) (ethtypes.Bloom, error)
+
+	BloomStatus() (uint64, uint64)
+
+	RPCFilterCap() int32
+	RPCLogsCap() int32
+	RPCBlockRangeCap() int32
+}
+
+// consider a filter inactive if it has not been polled for within deadline
+var deadline = 5 * time.Minute
+
+// filter is a helper struct that holds meta information over the filter type
+// and associated subscription in the event system.
+type filter struct {
+	typ      filters.Type
+	deadline *time.Timer // filter is inactive when deadline triggers
+	hashes   []common.Hash
+	crit     filters.FilterCriteria
+	logs     []*ethtypes.Log
+	s        *Subscription // associated subscription in event system
+}
+
+// PublicFilterAPI offers support to create and manage filters. This will allow external clients to retrieve various
+// information related to the Ethereum protocol such as blocks, transactions and logs.
+type PublicFilterAPI struct {
+	logger    log.Logger
+	clientCtx client.Context
+	backend   *backend.JSONRPCBackend
+	events    *EventSystem
+	filtersMu sync.Mutex
+	filters   map[rpc.ID]*filter
+}
+
+// NewFiltersAPI returns a new instance
+func NewFilterAPI(logger log.Logger, backend *backend.JSONRPCBackend, clientCtx client.Context, tmWSClient *rpcclient.WSClient) *PublicFilterAPI {
+	logger = logger.With("api", "filter")
+	api := &PublicFilterAPI{
+		logger:    logger,
+		clientCtx: clientCtx,
+		backend:   backend,
+		filters:   make(map[rpc.ID]*filter),
+		events:    NewEventSystem(logger, tmWSClient),
+	}
+
+	go api.timeoutLoop()
+
+	return api
+}
+
+// timeoutLoop runs every 5 minutes and deletes filters that have not been recently used.
+// Tt is started when the api is created.
+func (api *PublicFilterAPI) timeoutLoop() {
+	ticker := time.NewTicker(deadline)
+	defer ticker.Stop()
+
+	for {
+		<-ticker.C
+		api.filtersMu.Lock()
+		for id, f := range api.filters {
+			select {
+			case <-f.deadline.C:
+				f.s.Unsubscribe(api.events)
+				delete(api.filters, id)
+			default:
+				continue
+			}
+		}
+		api.filtersMu.Unlock()
+	}
+}
+
+// NewPendingTransactionFilter creates a filter that fetches pending transaction hashes
+// as transactions enter the pending state.
+//
+// It is part of the filter package because this filter can be used through the
+// `eth_getFilterChanges` polling method that is also used for log filters.
+//
+// https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_newPendingTransactionFilter
+/*
+func (api *PublicFilterAPI) NewPendingTransactionFilter() rpc.ID {
```

Comment:
@jasonsong0 
Does commented implementation works well? 
If not, it would be better to leave only method definition except implementation details.

And we should leave TODO comment for un-implemented apis.